src/lib/darkcoinClient.ts

```diff
@@ -154,6 +139,31 @@ export default class DarkcoinClient {
 
   // GObjects
 
+  /**
+   * The gobject prepare RPC prepares a governance object by signing and creating a collateral transaction.
+   * @param parentHash Hash of the parent object. usually the root node which has a hash of 0
+   * @param revision Object revision number (Always 0)
```

Comment:
Optional: If revision is always zero, I wonder if makes sense to either enforce this (by validating the input) or by omitting this parameter altogether. @jhodges10 Do you know why the point of a parameter that only takes the same value? I am not seeing the same message on https://dash-docs.github.io/en/developer-reference#gobject-prepare

Reply:
So revision always takes `0` because no other options were ever implemented according to @nmarley.

Reply:
Similar concept with `parent-hash` always being `0`, I think it's used in other parts of the `gobject` code for other reasons but not in the `gobject prepare` code. I've seen some skeletons in old branches of Dash that would have utilized the `parent-hash` differently that may make its way into my governance DIP.

Reply:
@jhodges10 thanks for clarification. Let's leave as it is for now, but if they really are going to always be zero for this method, then I propose removing it from the argument.

Reply:
I don't actually remember discussing this, but... revision should be `1`, not `0`. The parent-hash is a legacy which was intended to allow for a much more flexible gobject system which would work along with sentinel, but that's now being superceded by Evolution.

Clean-up (e.g. removing the arguments) would be nice, but easier said than done. Since the arguments are positional, DashCore and Sentinel would have to be updated at exactly the same time, otherwise the new DashCore wouldn't recognize the extra args from Sentinel or vice-versa. It's easier just to leave them until we remove this part completely.

Reply:
@nmarley thanks for chiming in. 

>  revision should be 1, not 0
gotcha. Either me or Jeff are going to fix it.

> Clean-up (e.g. removing the arguments) would be nice, but easier said than done

In this particular case, I was suggesting to remove it from this library as an argument, by changing the signature to something like:
```
public gobjectPrepare(parentHash: string, creationTime: number, gobjectData: string)
```
Then providing the revision as a hard-coded value. 

The goal of this library is to be as simple as possible.

Reply:
Ah, ok. I doubt those will ever change moving forward, so yeah, that makes sense. Eventually we'll remove the gobject system entirely.